### PR TITLE
Added more documentation

### PR DESCRIPTION
**Pull request by marcingrzejszczak**

### CODE REVIEW

File: docs/src/main/asciidoc/spring-cloud-sleuth.adoc [hunk 1]
```diff
@@ -273,10 +274,117 @@ the number of dropped spans will get increased.
...
+    - the Span will be tagged with that method's class name and the method name too
+
+==== Executor, ExecutorService and ScheduledExecutorService
+
+We're providing `LazyTraceExecutor`, `TraceableExecutorService` and `TraceableScheduledExecutorService`. Those implementations
+are creating Spans each time a new task is submitted, invoked or scheduled.
+
+=== Messaging
+
+To disable these features just set the `spring.sleuth.integration.enabled` property to `false`.
```
Comment (adriancole) [inline docs/src/main/asciidoc/spring-cloud-sleuth.adoc]: nit: I think you have at least 3 sentence variants of this instruction. For each section, 

// <20 word overview and how to disable
Foo instrumentation (adds X, tags Y, etc). To disable Foo instrumentation, set `spring.sleuth.foo.enabled` to false.

// if more info is available, detail it in a new paragraph

Ex 1. Spring Integration
> http://projects.spring.io/spring-integration/[Spring Integration] instrumentation creates spans for publish and subscribe  events. To disable Spring Integration instrumentation, set `spring.sleuth.integration.enabled` to false.

Ex 2. Zuul
> https://github.com/Netflix/zuul/[Zuul] instrumentation propagates trace information on outbound via http headers. To disable Zuul instrumentation, set `spring.sleuth.zuul.enabled` to false.

File: spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/documentation/SpringCloudSleuthDocTests.java [hunk 2]
```diff
@@ -61,8 +63,8 @@ public Sampler defaultSampler() {
 	}
 
 	// tag::span_name_annotation[]
-	@SpanName("calculateTax")
-	class TaxCountingRunnable implements Runnable {
+	@SpanName("processTax")
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/documentation/SpringCloudSleuthDocTests.java]: supernit: was this rename intentional? I think calculateTax is a decent name and also doesn't incidentally join the word Process with Runnable

**Comment by adriancole** [comment]
looks like progress. thanks!

File: spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/documentation/SpringCloudSleuthDocTests.java (hunk 2 above)
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/documentation/SpringCloudSleuthDocTests.java]: yeah that was exactly why I did it. Ok I'll rename it then back to what it was.

**Comment by marcingrzejszczak** [comment]
Done via https://github.com/spring-cloud/spring-cloud-sleuth/commit/6b6f40aed2e5ba3d1983c265084e9465231e8592
